### PR TITLE
test: mheap: add testcase

### PR DESCRIPTION
**Pull request by yingmingx**
This is an intergration testcase for mem_heap.
Add an testcase to verify that multiple threads
can share the same heap space without interfering
with each other.

Signed-off-by: Ying ming <mingx.ying@intel.com>

### CODE REVIEW

**Review by LeiW000 — CHANGES_REQUESTED** [review]

**Comment by LeiW000** [comment]
I think that we also need to malloc/free difference size of memory several times with multi-threads.

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c [hunk 1]
```diff
@@ -44,3 +51,49 @@ void test_mheap_malloc_align4(void)
...
+					 K_PRIO_PREEMPT(1), 0, K_NO_WAIT);
+	}
+
+	for (int i = 0; i < THREAD_NUM; i++) {
+		k_sem_take(&sync_sema, K_FOREVER);
+	}
+
+	for (int i = 0; i < THREAD_NUM; i++) {
+		/* verify free mheap in main thread */
+		k_free(block[i]);
```
[the remark this answers is not in the document]
Comment (andyross) [reply tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c:96]: k_free(NULL) is a documented noop, just like free(NULL) is in the C standard.  This is fine as long as block[i] always holds the return value from k_malloc().

**Review by andyross — APPROVED** [review]
Looks fine.  FWIW: if you really want to test for race conditions you're going to need to be doing thousands of allocations in a preemptable (and ideally SMP) context and not just one per thread.  But this isn't necessarily a stress test.

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (LeiW000) [reply tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c:96]: @andyross, thanks. I was not aware of it.

**Review by LeiW000 — APPROVED** [review]

**Pull request merged**